lib/ansible/modules/cloud/memset/memset_memstore_user.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Simon Weald <ansible@simonweald.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: memset_memstore_user
+author: "Simon Weald (@glitchcrab)"
+version_added: "2.9"
+short_description: Manage Memstore users
+notes:
+  - Create/delete and manage the status of Memstore users.
+  - An API key generated via the Memset customer control panel is needed with
+    the following minimum scope - I(memset.user_list), I(memstore.user_info),
+    I(memstore.user_enable), I(memstore.user_disable), I(memstore.user_delete),
+    I(memstore.user_create).
+description:
+    - Create/delete and manage the status of Memstore users.
+options:
+    state:
+        default: present
+        type: str
+        description:
+            - Indicates desired state of resource.
+        choices: [ absent, present ]
+    api_key:
+        required: true
+        type: str
+        description:
+            - The API key obtained from the Memset control panel.
+    username:
+        required: true
+        type: str
+        description:
+            - The name of the user to manage.
+        aliases: [ name ]
+    memstore:
+        required: true
+        type: str
+        description:
+            - The Memstore product the user belongs to.
+    enabled:
+        default: false
+        type: bool
+        description:
+            - Whether the user is enabled or disabled. Defaults to false for security.
+    password:
+        type: str
+        description:
+            - A password for the user. Required when the user is present.
+            - C(update_password) must be C(always) for an existing user's password to be updated after initial
+              creation.
+    update_password:
+        default: on_create
+        type: str
+        description:
+            - Set the user's password. By default this always happens on user creation as C(update_password)
+              defaults to C(on_create).
+            - Alternatively, this can be set to C(always) which will ensure the password is set on each task
+              invocation. The underlying API does not expose whether the password has actually been changed, so
+              using C(always) will result in the task always returning changed.
+            - This parameter requires C(password) to be set.
```

Comment:
Shouldn't this have `choices`? Or what do you want to do if the user enters a random string?

Reply:
Good point, I completely missed that. I've just pushed a fix.

Reply:
You also need to add the choices in the `argument_spec`.

Reply:
yup, sorted that now too.